src/operators/mergeMap-support.ts

```diff
@@ -49,36 +51,39 @@ export class MergeMapSubscriber<T, R, R2> extends OuterSubscriber<T, R> {
       this.buffer.push(value);
     }
   }
-  
+
   _innerSub(ish: any, value: T, index: number) {
     this.add(subscribeToResult<T, R>(this, ish, value, index));
   }
-  
+
   _complete() {
     this.hasCompleted = true;
-    if(this.active === 0 && this.buffer.length === 0) {
+    if (this.active === 0 && this.buffer.length === 0) {
       this.destination.complete();
     }
   }
-  
+
   notifyNext(innerValue: R, outerValue: T, innerIndex: number, outerIndex: number) {
     const { destination, resultSelector } = this;
-    if(resultSelector) {
-      const result = tryCatch(resultSelector)(innerValue, outerValue, innerIndex, outerIndex);
-      if(result === errorObject) {
+    if (resultSelector) {
+      const result = tryCatch(resultSelector)(
+        innerValue, outerValue, innerIndex, outerIndex
+      );
+      if (result === errorObject) {
         destination.error(errorObject.e);
       } else {
         destination.next(result);
       }
+    } else {
+      destination.next(innerValue);
     }
-    destination.next(innerValue);
   }
```

Comment:
This above wasn't just a refactor, I recall solving one of the tests failing by including this `else`, because otherwise there would be two calls to `destination.next()`.